locator/tablets.hh

```diff
@@ -216,6 +218,18 @@ public:
         return _tablets;
     }
 
+    /// Calls a given function for each tablet in the map in token ownership order.
+    template <typename Func>
+    requires std::invocable<Func, tablet_id, const tablet_info&>
+    future<> for_each_tablet(Func func) const {
+        std::optional<tablet_id> tid = first_tablet();
+        for (const tablet_info& ti : tablets()) {
+            co_await seastar::coroutine::maybe_yield();
+            co_await futurize_invoke(func, *tid, ti);
+            tid = next_tablet(*tid);
```

Comment:
tablet_map is immutable? Otherwise we need to know which lock needs to be held.

Reply:
It is not immutable. When it's part of group0 state machine, it's locked by group0_guard.

Reply:
ok

Reply:
Well, no, it is actually immutable.

Reply:
Even ok'er